src/tests/utests/xi_utest_event_dispatcher.c

```diff
@@ -60,11 +61,68 @@ xi_state_t proc_loop( xi_event_handle_arg1_t a )
 
     if ( *( ( uint32_t* )a ) > 0 )
     {
-        xi_evtd_execute_in( evtd_g_i, evtd_handle_g, 1 );
+        xi_evtd_execute_in( evtd_g_i, evtd_handle_g, 1, NULL );
     }
     return 0;
 }
 
+void test_time_overflow_function( void )
+{
+    evtd_g_i = xi_evtd_create_instance();
+
+    tt_assert( NULL != evtd_g_i );
+
+    /* counter that will be used by this test */
+    uint32_t counter = 0;
+
+    const xi_time_t time_max  = LONG_MAX;
+    const xi_time_t time_hole = 10;
+
+    printf( "%ld\r\n", time_max );
+
+    evtd_g_i->current_step = time_max - time_hole;
+
+    /* now let's add some events to be executed */
+    /* we will add some events before the MAX( xi_time_t ) */
+    {
+        size_t i = 0;
+        for ( ; i < time_hole; ++i )
+        {
+            const xi_state_t ret_state = xi_evtd_execute_in(
+                evtd_g_i, xi_make_handle( &continuation1_1, &counter ), i, NULL );
+
+            xi_debug_format( "ret_state = %d", ret_state );
+
+            tt_int_op( XI_STATE_OK, ==, ret_state );
+        }
+    }
+
+    /* and some after the MIN( xi_time_t ) in order to check the reaction on time
+     * overlap */
+    {
+        size_t i = 0;
+        for ( ; i < time_hole; ++i )
+        {
+            const xi_state_t ret_state = xi_evtd_execute_in(
+                           evtd_g_i, xi_make_handle( &continuation1_1, &counter ),
+                           time_hole + i, NULL );
+            tt_int_op( XI_STATE_OK, ==, ret_state );
+        }
+    }
+
+    /* let's execute first half */
+    xi_evtd_step( evtd_g_i, time_max - ( time_hole / 2 ) );
+
+    /* expectation is that only the half of the events added before the MAX( xi_time_t
+     * ) will be executed */
```

Comment:
an unfortunate line break in the comment.  Not sure if it's possible to fix this.